lib/js-yaml/dumper.js

```diff
@@ -738,7 +739,7 @@ function writeNode(state, level, object, block, compact, iskey) {
       if (block && (0 !== Object.keys(state.dump).length)) {
         writeBlockMapping(state, level, state.dump, compact);
         if (duplicate) {
-          state.dump = '&ref_' + duplicateIndex + (0 === level ? '\n' : '') + state.dump;
+          state.dump = '&ref_' + duplicateIndex + state.dump;
```

Comment:
Ah, I remembered. It's for anchors on top-level node. So the question stands: Why is this removed?